### ISSUE TITLE
Mandarin label. the target is [0,0,0,0,........0]

### ISSUE THREAD

**Issue by minushuang** [SeanNaren/deepspeech.pytorch#328]
I have generated the labels.json as ["_", "语", "音", "识", "别",...,"百", "度"]. 

then I modified the train.py code to decode the labels.json from utf-8 to unicode
`with open(args.labels_path) as label_file:
labels = str(''.join(json.load(label_file)))
 labels = labels.decode('utf-8')`

and to see the train-data details, I printed the inputs,  targets, input_percentages,  target_sizes and the length correspond to them.

I got the result of targets = tensor([], dtype=torch.int32),  len(targets) = 0, and the target_sizes = [0,0,...,0].

there must be some errors with my code when I load the Chinese character datasets or labels.json.

How can fix this ?

**Comment by miguelvr** [comment]
Maybe it's the encoding? You should check what is happening in `SpectrogramDataset`, i.e. how is the labels map being constructed and what is happening in `parse_transcript`. These should work fine, but it may help you figure out what's wrong with your target

**Comment by minushuang** [comment]
@miguelvr 
Thanks for reply.
another question about labels for Mandarin.
I have added the blank '_'  in labels.json.
Dose it is necessary to add a space '  ' for training the Mandarin models.

**Comment by miguelvr** [comment]
@minushuang does mandarin have spaces in it? If it does, then yes.

The blank '_' char is a special symbol used by the CTC to infer double symbols:

Example:
if the output of the model was `hhheellllo` then the CTC would output `helo`, in otder to output `hello` the model would have to predict `hhheell-llo`
